### PR TITLE
Add example visualizing RRT*

### PR DESCRIPTION
**Pull request by 02alexander**
### What

Adds an python example that visualizes the RRT* pathfinding algorithm.  There isn't a screenshot of the viewer in the `README.md` yet but it will be added by linking to the image in this pull request.  
I have tested the example in the native viewer in  and the web viewer but I'm not sure if that's what  "[ ] I have tested the web demo" means.

rrt-star-screenshot

### Checklist
* [x] I have read and agree to [Contributor Guide](https://github.com/rerun-io/rerun/blob/main/CONTRIBUTING.md) and the [Code of Conduct](https://github.com/rerun-io/rerun/blob/main/CODE_OF_CONDUCT.md)
* [x] I've included a screenshot or gif (if applicable)
* [x] I have tested the web demo (if applicable):
* [x] The PR title and labels are set such as to maximize their usefulness for the next release's CHANGELOG

- [PR Build Summary](https://build.rerun.io/pr/5214)
- [Docs preview](https://rerun.io/preview/48476971401c69710e7c0ffd1a45b97206d2ef9b/docs) 
- [Examples preview](https://rerun.io/preview/48476971401c69710e7c0ffd1a45b97206d2ef9b/examples) 
- [Recent benchmark results](https://build.rerun.io/graphs/crates.html)
- [Wasm size tracking](https://build.rerun.io/graphs/sizes.html)

### CODE REVIEW

**Review by emilk — COMMENTED** [review]
Very cool 🤩

**Review by emilk — APPROVED** [review]
I'm no Python coder, but looks good to me!

File: examples/python/rrt-star/main.py [hunk 1]
```diff
@@ -0,0 +1,280 @@
...
+                rr.log(
+                    "map/path",
+                    rr.LineStrips2D(segments, radii=0.002, colors=[0, 255, 255, 255]),
+                )
+
+    return path
+
+
+def main() -> None:
+    parser = argparse.ArgumentParser(description="Example of using the Rerun visualizer")
```
Comment (Wumpf) [inline examples/python/rrt-star/main.py:248]: we can do a bit better than that :). Please describe here what the application does

File: examples/python/rrt-star/main.py [hunk 2]
```diff
@@ -0,0 +1,280 @@
+#!/usr/bin/env python3
+"""
+Visualizes the path finding algorithm RRT* in a simple environment.
```
Comment (Wumpf) [inline examples/python/rrt-star/main.py:3]: I'm new to this, what is RRT* and how does it work :)
We should add a link with more learning material here and in the readme

**Review by Wumpf — CHANGES_REQUESTED** [review]
looking cool, but I for one need more docs!

**Comment by Wumpf** [comment]
> I have tested the example in the native viewer in and the web viewer but I'm not sure if that's what "[ ] I have tested the web demo" means.

This is only relevant for changes to the viewer and api. We usually want to make sure that the online web demo the ci builds still works with those changes. For a new example you can ignore this :)

File: examples/python/rrt-star/main.py (hunk 2 above)
Comment (02alexander) [reply examples/python/rrt-star/main.py:3]: I have added a brief description of the algorithm in a `rr.TextDocument` with links to relevant entity paths. Here is the screenshot: 
rrt-star-screenshot

**Comment by 02alexander** [comment]
The link to the screenshot of the example seems to be acting weird. I don't have `gcloud` access yet so I can't use `just upload`. So I tried linking to the images I uploaded in this PR but it's looks like the urls are changing.

File: examples/python/rrt-star/main.py (hunk 1 above)
Comment (02alexander) [reply examples/python/rrt-star/main.py]: Oops, I forgot to change that after copying the template. Fixed now.

**Review by Wumpf — APPROVED** [review]
nice, much better

**Pull request merged**